### ISSUE TITLE
426 no sec-websocket-protocols requested supported

### ISSUE THREAD

**Issue by lanstar2** [simonhamp/echo#1]
426 no sec-websocket-protocols requested supported

**Comment by simonhamp** [comment]
@lanstar2 could you elaborate?

**Comment by francislavoie** [comment]
Aha! I just ran into this issue and I just figured out what this issue was referring to... horrible job by @lanstar2 of explaining what he meant (literally no details), but this is the problem:

On Firefox, websockets fail. A `426 Upgrade Required` is returned.

The reason is that the `Sec-WebSocket-Protocol` header is set to `undefined`. The reason is the following:

https://github.com/simonhamp/echo/blob/ratchet-connector/src/connector/ratchet-connector.ts#L28

On the above line, `this.options.protocols` is passed to `new Websocket`. On Firefox, for some weird reason, it will actually take this undefined variable _as a string_ and feeds it to the above header. Stupid, stupid behaviour.

Easy fix though, we just need to wrap that call and avoid using that parameter if `this.options.protocols` is undefined. I'll make a quick PR.
